engine/apps/labels/alert_group_labels.py

```diff
@@ -99,51 +95,52 @@ def _custom_labels(
         ).only("id", "name")
     }
 
-    rendered_labels = {}
+    result_labels = {}
     for label in alert_receive_channel.alert_group_labels_custom:
-        key_id, value_id, template = label
-
-        if key_id in label_key_names:
-            key = label_key_names[key_id]
-        else:
-            logger.warning("Label key cache not found. %s", key_id)
-            continue
-
-        if value_id:
-            if value_id in label_value_names:
-                rendered_labels[key] = label_value_names[value_id]
-            else:
-                logger.warning("Label value cache not found. %s", value_id)
-                continue
-        else:
-            try:
-                rendered_labels[key] = apply_jinja_template(template, raw_request_data)
-            except (JinjaTemplateError, JinjaTemplateWarning) as e:
-                logger.warning("Failed to apply template. %s", e.fallback_message)
-                continue
-
-    labels = {}
-    for key in rendered_labels:
-        value = rendered_labels[key]
+        label = _apply_dynamic_label_entry(label, label_key_names, label_value_names, raw_request_data)
+        if label:
+            key, value = label
+            result_labels[key] = value
 
-        # check value length
-        if len(value) == 0:
-            logger.warning("Template result value is empty. %s", value)
-            continue
-
-        if len(value) > MAX_VALUE_NAME_LENGTH:
-            logger.warning("Template result value is too long. %s", value)
-            continue
+    return result_labels
 
-        labels[key] = value
 
-    return labels
+def _apply_dynamic_label_entry(
+    label: "AlertReceiveChannel.DynamicLabelsEntryDB", keys, values, payload
```

Comment:
it's old logic of applying custom labels, but without "static" values - they now live in integration associated labels exclusively. Also It calls _validate_templated_value to fix actual big with None values.